File: modules/scripting/swig/php/test/test_swig_api.php
#!/usr/bin/php
<?php

include("oescriptimpl.php");
assert_options(ASSERT_ACTIVE, true);
assert_options(ASSERT_WARNING, true);
assert_options(ASSERT_BAIL, true);
assert_options(ASSERT_CALLBACK, 'assert_callback');

function assert_callback( $script, $line, $message ) {
    echo 'error: ', $script,': line', $line,":\n";
    exit(1);
}

$signals = new_OesSigHandler("mysigfile.txt", "oesphp", "info");
assert($signals);

$dispatcher = new_OesDispatcher(1, "mydispatcher");
assert($dispatcher);
OesDispatcher_start($dispatcher);

$tdispatcher = new_OesThreadDispatcher(5);
assert($tdispatcher);
OesThreadDispatcher_start($tdispatcher);

$store = new_OesStore(3, "data", 1);
assert($store);

$kernel = new_OesKernel($dispatcher, $tdispatcher, $store, null, null, null, null);
assert($kernel);

#start socket server
$net = new_OesNet($dispatcher, 0, null, null);
assert($net);
$json = new_OesServer($dispatcher, "oejson://0.0.0.0:9555", $net, $store);
assert($json);
OesServer_addKernel($json, "myspace", $kernel);

#shutdown

?>

File: modules/scripting/swig/ruby/test/test_swig_api.rb
#!/usr/bin/ruby
require 'test/unit'
require 'oescriptimpl'

class TestRubyMod < Test::Unit::TestCase

  def test_runme
  	signals = Oescriptimpl::OesSigHandler.new("mysigfile.pid", "oesrb", "info")
	assert_not_nil(signals)

	dispatcher = Oescriptimpl::OesDispatcher.new(1, "mydispatcher")
	assert_not_nil(dispatcher)
	dispatcher.start()

	tdispatcher = Oescriptimpl::OesThreadDispatcher.new(4)
	assert_not_nil(tdispatcher)
	tdispatcher.start()

	store = Oescriptimpl::OesStore.new(3, "data", 1)
	assert_not_nil(store)

	kernel = Oescriptimpl::OesKernel.new(dispatcher, tdispatcher, store,
    nil, nil, nil, nil)
	assert_not_nil(kernel)

	#start socket server
	net = Oescriptimpl::OesNet.new(dispatcher, false, nil, nil)
	assert_not_nil(net)
	json = Oescriptimpl::OesServer.new(dispatcher, "oejson://0.0.0.0:7777",net, store)
	assert_not_nil(json)
    json.addKernel("myspace", kernel)

  end
end

